pds_doi_service/api/controllers/dois_controller.py

```diff
@@ -114,13 +114,21 @@ def _records_from_dois(dois, node=None, submitter=None, osti_label=None):
     """
     records = []
 
+    list_action = DOICoreActionList(db_name=_get_db_name())
+
     for doi in dois:
+        # Pull info from transaction database so we can get the most accurate
+        # info for the DOI
+        list_kwargs = {'lidvid': doi.related_identifier}
+        list_result = json.loads(list_action.run(**list_kwargs))[0]
+
         records.append(
             DoiRecord(
-                doi=doi.doi, lidvid=doi.related_identifier, title=doi.title,
+                doi=doi.doi or list_result['doi'],
```

Comment:
The Zen of Python (`import this`) says "explicit > implicit" which suggests:

```python
doi=doi.doi if doi.doi is not None and len(doi.doi) > 0 else list_results['doi'],
```

since this protects against `doi.doi` being a random value like boolean `True`. But I will admit, I like the concision of `param=x or y`.